ddtrace/internal/atexit.py

```diff
@@ -0,0 +1,59 @@
+# -*- encoding: utf-8 -*-
+"""
+An API to provide atexit functionalities
+"""
+import atexit
+import logging
+import typing
+
+
+log = logging.getLogger(__name__)
+
+
+if hasattr(atexit, "unregister"):
+    register = atexit.register
+    unregister = atexit.unregister
+else:
+    # Hello Python 2!
+    _registry = []  # type: typing.List[typing.Callable[[], None]]
+
+    def _ddtrace_atexit():
+        # type: (...) -> None
+        """Wrapper function that calls all registered function on normal program termination"""
+        global _registry
+
+        # DEV: we make a copy of the registry to prevent hook execution from
+        # introducing new hooks, potentially causing an infinite loop.
+        for hook in list(_registry):
+            try:
+                hook()
+            except Exception:
+                # Mimic the behaviour of Python's atexit hooks.
+                log.exception("Error in atexit hook %r", hook)
+
+    atexit.register(_ddtrace_atexit)
+
+    def register(
+        func,  # type: typing.Callable[..., typing.Any]
+        *args,  # type: typing.Any
+        **kwargs,  # type: typing.Any
```

Comment:
Somehow this seems to be invalid syntax on Python 2.7 and 3.5 🤷

Reply:
😢